drivers/SmartThings/matter-sensor/src/init.lua

```diff
@@ -56,6 +62,27 @@ local function set_field_for_endpoint(device, field, endpoint, value, additional
   device:set_field(string.format("%s_%d", field, endpoint), value, additional_params)
 end
 
+local BOOLEAN_DEVICE_TYPE_INFO = {
+  ["RAIN_SENSOR"] = { id = 0x0044, sensitivity_preference = "rainSensitivity" },
+  ["WATER_FREEZE_DETECTOR"] = { id = 0x0041, sensitivity_preference = "freezeSensitivity" },
+  ["WATER_LEAK_DETECTOR"] = { id = 0x0043, sensitivity_preference = "leakSensitivity" },
+  ["CONTACT_SENSOR"] = { id = 0x0015, sensitivity_preference = "N/A" },
+}
+
+local function set_device_type_per_endpoint(driver, device)
+  for _, ep in ipairs(device.endpoints) do
+      for _, dt in ipairs(ep.device_types) do
+          for dt_name, info in pairs(BOOLEAN_DEVICE_TYPE_INFO) do
+              if dt.device_type_id == info.id then
+                  device:set_field(dt_name, ep.endpoint_id)
+                  device:send(clusters.BooleanStateConfiguration.attributes.DefaultSensitivityLevel:read(device, ep.endpoint_id))
```

Comment:
You can merge these two messages into one before sending

Reply:
moot point now, remove defaultsensitivity read.